account-keys/src/view_key.rs

```diff
@@ -1,54 +0,0 @@
-// Copyright (c) 2018-2021 The MobileCoin Foundation
-
-//! CryptoNote-style View Key.
-//!
-//! A view key (a,B) contains half of a user's private information, and is used
-//! to identify transaction outputs sent to the user, and to view the amounts of
-//! those outputs.
-
-use core::hash::{Hash, Hasher};
-use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
-use prost::Message;
-
-/// The user's (a,B) keys.
-#[derive(Clone, Message)]
-pub struct ViewKey {
-    /// The user's private key `a`.
-    #[prost(message, required, tag = 1)]
-    pub view_private_key: RistrettoPrivate,
-    /// The user's public key `B`
-    #[prost(message, required, tag = 2)]
-    pub spend_public_key: RistrettoPublic,
-}
```

Comment:
It's actually incorrect that this isn't being used currently, at least one outside consumer of Core are using (full service):
https://github.com/mobilecoinofficial/full-service/blob/3c9c1b816b61f2cd6a09574dd8fcfddd511bc28c/full-service/src/db/assigned_subaddress.rs#L153

Eliminating this might cause regressions in other parts of the ecosystem

Reply:
Perhaps we need to put tests in that run other parts of the ecosystem?

Reply:
I think full service doesn't actually need to use it there, the only usage is a few lines later:

```
            let txo_subaddress_spk: RistrettoPublic = recover_public_subaddress_spend_key(
                &account_view_key.view_private_key,
                &tx_out_target_key,
                &tx_public_key,
            );
```

So instead of calling `account_key.view_key()` they can just call `account_key.view_private_key()`, and the code will work the same as before.

Happy to work with full service team on this. Brian was under the impression earlier that `ViewKey` is never needed, so I'll be surprised if there are blockers.

Reply:
that is the only hit I see for `ViewKey` or `.view_key()` in all of full-service

Reply:
Ya, that seems like the case. There's not much test coverage on Full Service right now and increasing usage of it (binance, etc.), we probably want to put some full service tests in the CI flow (under like.. the FOG SLA or something)